lib/boom_notifier/notification_sender.ex

```diff
@@ -0,0 +1,124 @@
+defmodule BoomNotifier.NotificationSender do
+  @moduledoc false
+
+  # This GenServer is responsible for sending the notifications in background.
+
+  require Logger
+  use GenServer
+
+  alias BoomNotifier.ErrorInfo
+  alias BoomNotifier.ErrorStorage
+
+  # Client API
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  def async_notify(notifier, occurrences, options) do
+    GenServer.cast(__MODULE__, {:notify, notifier, occurrences, options})
+  end
+
+  def async_trigger_notify(settings, error_info) do
+    GenServer.cast(__MODULE__, {:trigger_notify, settings, error_info})
+  end
+
+  def notify(notifier, occurrences, options) do
+    spawn_link(fn ->
+      notifier.notify(occurrences, options)
+    end)
+  end
+
+  def notify_all(settings, error_info) do
+    notification_trigger = Keyword.get(settings, :notification_trigger, :always)
+    occurrences = Map.put(error_info, :occurrences, ErrorStorage.get_error_stats(error_info))
+
+    ErrorStorage.reset_accumulated_errors(notification_trigger, error_info)
+
+    BoomNotifier.Api.walkthrough_notifiers(
```

Comment:
Is there an actual benefit of doing `BoomNotifier.Api.walkthrough_notifiers` instead of just `BoomNotifier.walkthrough_notifiers`? Or it's just to separate the macro definition from the rest of the functions?

@germanbottosur Do you know if there's a convention for separating this type of concern?

Reply:
there's no real benefit but offloading BoomNotifier module